ircd/ircd_relay.c

```diff
@@ -355,6 +359,17 @@ void relay_directed_notice(struct Client* sptr, char* name, char* server, const
       (!EmptyString(host) && 0 != match(host, cli_user(acptr)->host)))
     return;
 
+  /* Apply the same logic to NOTICE that is applied to PRIVMSG: only
+   * allow services to receive /notice nick@server.undernet.org notices.
+  */
+  if (!(acptr = FindUser(name)) || !MyUser(acptr) ||
+      (!EmptyString(host) && 0 != match(host, cli_user(acptr)->host)) ||
+      !IsChannelService(acptr))
```

Comment:
IsChannelService() should be checked before the call to match(), both as a minor speed optimization and to prevent information leakage from how long match() takes to run.

Otherwise +1.

Reply:
Done. See commit [9f28062](https://github.com/UndernetIRC/ircu2/pull/31/commits/9f28062b3d6e5b665c7bb742e75c3da8b1ec01bd).